### PR TITLE
feat(lint): enforce jsdoc of widgets + exported functions in core closes #30

### LINKED ISSUE #30: Enforce proper documentation of core components
**Is your feature request related to a problem? Please describe.**
There is no automatic check that helps developers write api documentation of their core component.

**Describe the solution you'd like**
Use eslint plugin https://github.com/gajus/eslint-plugin-jsdoc with some smart configuration to enforce proper api documentation.

### PR DESCRIPTION
**Pull request by quentinderoubaix**

### CODE REVIEW

File: core/package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
 		"test:coverage": "vitest run --coverage"
 	},
 	"dependencies": {
-		"@amadeus-it-group/tansu": "0.0.22"
+		"@amadeus-it-group/tansu": "0.0.22",
+		"remove": "^0.1.5"
```
Comment (divdavem) [inline core/package.json:30]:
```suggestion
```

File: core/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+			"error",
+			{
+				"definedTags": ["remarks", "defaultValue"]
+			}
+		],
+		"jsdoc/require-jsdoc": [
+			"error",
+			{
+				"publicOnly": true,
+				"contexts": ["ExportNamedDeclaration TSInterfaceDeclaration > TSInterfaceBody > TSPropertySignature"]
```
Comment (divdavem) [inline core/.eslintrc.json:15]:
```suggestion
				"contexts": [
					"TSInterfaceDeclaration > TSInterfaceBody > TSPropertySignature",
					"TSInterfaceDeclaration > TSInterfaceBody > TSMethodSignature",
					"TSTypeAliasDeclaration > TSTypeLiteral > TSPropertySignature",
					"TSTypeAliasDeclaration > TSTypeLiteral > TSMethodSignature"
				]
```

File: core/lib/rating.ts [hunk 3]
```diff
@@ -96,9 +102,21 @@ export interface RatingProps extends RatingCommonPropsAndState {
 }
 
 export interface RatingState extends RatingCommonPropsAndState {
+	/**
+	 * the aria value of the rating
+	 */
 	ariaValueText: string;
+	/**
+	 * the value of the rating
```
Comment (divdavem) [inline core/lib/rating.ts:110]:
```suggestion
	 * the visible value of the rating (it changes when hovering over the rating even though the real value did not change)
```

File: core/lib/transitions/simpleClassTransition.ts [hunk 4]
```diff
@@ -2,14 +2,32 @@ import {createCSSTransition} from './cssTransitions';
...
+	/**
+	 * a list of classes to add to the collapsable element while transitioning
+	 */
 	animationPendingClasses?: string[];
+	/**
+	 * a list of classes to add to the collapsable element while transitioning towards the show state
+	 */
 	animationPendingShowClasses?: string[];
+	/**
+	 * a list of classes to add to the collapsable element while transitioning towards the hidden state
```
Comment (divdavem) [inline core/lib/transitions/simpleClassTransition.ts:14]:
```suggestion
	 * a list of classes to add to the collapsable element while transitioning towards the hide state
```
(for consistency with the previous item, either `shown` and `hidden` or `show` and `hide`)

File: core/lib/types.ts [hunk 5]
```diff
@@ -8,20 +8,43 @@ export interface Widget<
 	Actions extends object = object,
 	Directives extends object = object
 > {
+	/**
+	 * the reactive state of the widget
+	 */
 	state$: ReadableSignal<State>;
+	/**
+	 * the different stores of the widget, all reactive
```
Comment (divdavem) [inline core/lib/types.ts:16]: Maybe you can mention that the `state$` store contains all the values of the stores present in this object.

File: core/lib/types.ts [hunk 6]
```diff
@@ -8,20 +8,43 @@ export interface Widget<
...
+	 */
 	stores: {[K in keyof State as `${K & string}$`]: ReadableSignal<State[K]>};
 
 	/**
 	 * Modify the parameter values, and recalculate the stores accordingly
 	 */
 	patch(parameters: Partial<Props>): void;
+	/**
+	 * the directives that can be used on associated elements to react to the widget.
```
Comment (divdavem) [inline core/lib/types.ts:25]: Directives to be used on html elements in the template of the widget itself or in the slots.

File: core/package.json (hunk 1 above)
Comment (quentinderoubaix) [reply core/package.json:30]: done thx !

File: core/.eslintrc.json (hunk 2 above)
Comment (quentinderoubaix) [reply core/.eslintrc.json:15]: done, thx !

File: core/lib/rating.ts (hunk 3 above)
Comment (quentinderoubaix) [reply core/lib/rating.ts:110]: done, thx !

File: core/lib/types.ts (hunk 5 above)
Comment (quentinderoubaix) [reply core/lib/types.ts:16]: done, thx !

**Review by divdavem — APPROVED** [review]

**Pull request merged**